strawberry/relay/fields.py

```diff
@@ -81,7 +81,9 @@ async def resolve_async(
         # async extensions.
         return await retval if inspect.isawaitable(retval) else retval
 
-    def get_node_resolver(self, field: StrawberryField):  # noqa: ANN201
+    def get_node_resolver(
+        self, field: StrawberryField
+    ) -> Callable[[Info, GlobalID], Union[Node, None, Awaitable[Union[Node, None]]]]:
```

Comment:
Might be worth adding a `MaybeAwaitable` type alias in the future for readability.

Reply:
graphql-core has it:

```python
from graphql.pyutils import AwaitableOrValue
```

Using it this could be written as `Callable[[Info, GlobalID], AwaitableOrValue[Optional[Node]]]`